internal/unit-info-viewer/skills-progress.lua

```diff
@@ -0,0 +1,181 @@
+--@ module=true
+
+local utils = require("utils")
+local widgets = require('gui.widgets')
+local overlay = require('plugins.overlay')
+
+local view_sheets = df.global.game.main_interface.view_sheets
+
+local function get_skill(id)
+    return utils.binsearch(
+        df.unit.find(view_sheets.viewing_unid[0]).status.current_soul.skills,
+        view_sheets.unit_skill[id],
+        "id"
+    )
+end
+
+SkillProgressOverlay=defclass(SkillProgressOverlay, overlay.OverlayWidget)
+SkillProgressOverlay.ATTRS {
+    desc="Display progress bars for learning skills!",
+    default_pos={x=-43,y=20},
+    default_enabled=true,
+    viewscreens= {
+        'dwarfmode/ViewSheets/UNIT/Skills/Labor',
+        'dwarfmode/ViewSheets/UNIT/Skills/Combat',
+        'dwarfmode/ViewSheets/UNIT/Skills/Social',
+        'dwarfmode/ViewSheets/UNIT/Skills/Other',
+
+        'dungeonmode/ViewSheets/UNIT/Skills/Labor',
+        'dungeonmode/ViewSheets/UNIT/Skills/Combat',
+        'dungeonmode/ViewSheets/UNIT/Skills/Social',
+        'dungeonmode/ViewSheets/UNIT/Skills/Other',
+    },
+    frame={w=54, h=67},
```

Comment:
Note that the frame height will need to be dynamically calculated as the screen is resized. For an initial value for `h`, choose the height when the screen is a minimum size (114x46). If you choose too large of a height, the overlay widget can get permanently pushed out of position. I expect you'll get something around `h=20`.